frontend/src/hooks/fxaFlowTracker.ts

```diff
@@ -0,0 +1,71 @@
+import { useEffect, useState } from "react";
+import { event as gaEvent, EventArgs } from "react-ga";
+import { IntersectionOptions, useInView } from "react-intersection-observer";
+import { useRuntimeData } from "./api/runtimeData";
+
+export type FxaFlowTrackerArgs = Omit<
+  EventArgs,
+  "action" | "nonInteraction"
+> & { entrypoint: string };
+
+/**
+ * This is similar to {@see useGaViewPing}, but this also gives you a `flowId`
+ * and `flowBeginTime` that you can send to FxA to track sign-in/-up flows.
+ */
+export function useFxaFlowTracker(
+  args: FxaFlowTrackerArgs | null,
+  options?: IntersectionOptions
+) {
+  const runtimeData = useRuntimeData();
+  const [ref, inView] = useInView({ threshold: 1, ...options });
+  const [flowData, setFlowData] = useState<{
+    flowId: string;
+    flowBeginTime: string;
+  }>();
+
+  // If the API hasn't responded with the environment variables by the time this
+  // hook fires, we assume we're dealing with the production instance of FxA:
+  const fxaOrigin =
+    runtimeData.data?.FXA_ORIGIN ?? "https://accounts.firefox.com";
+
+  useEffect(() => {
+    if (args === null || !inView) {
+      return;
+    }
+    gaEvent({
+      ...args,
+      action: "View",
+      nonInteraction: true,
+    });
```

Comment:
praise: thank you for preserving the event record into our own GA so we can continue to see these in our own GA reports too.